src/Microsoft.Azure.VirtualKubelet.Edge.Provider/Dockerfile

```diff
@@ -1,9 +1,19 @@
-FROM microsoft/dotnet:2.0.0-runtime
-
+FROM microsoft/dotnet:sdk AS build-env
 WORKDIR /app
 
-COPY ./ /app
+# Copy csproj and restore as distinct layers
+ADD ./ ./
+
+RUN dotnet restore
 
-EXPOSE 5000
```

Comment:
EXPOSE 5000 should not be removed entirely. Otherwise the health and readiness checks in Kubernetes will be broken.